### PR TITLE
Add hacking guideline

### PR DESCRIPTION
**Pull request by mwjames**
This PR is made in reference to: #

This PR addresses or contains:

This PR includes:
- [ ] Tests (unit/integration)
- [ ] CI build passed

### CODE REVIEW

File: docs/architecture/coding.conventions.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- `Constants` are written ALL_CAPS with underscores as internal separator, and are to be prefixed with "SMW_" when used globally (e.g. in "define('SMW_FOO', 1)"), class constants are free from this convention.
+
+### Code layout and indenting
+
+In general, code layout is guided by the [MediaWiki coding conventions](http://www.mediawiki.org/wiki/Manual:Coding_conventions).  Please be sure to read this document.
+
+- Do not use a closing "?>" tag in your source files. It is a source for errors that is not needed in files.
+- Document all your code (see source documentation for details)
+- Avoid single lines of code becoming too long.
+- '''Indenting''' of program blocks is done with tabulators, not with spaces.All program blocks are indented.
```
Comment (Nikerabbit) [inline docs/architecture/coding.conventions.md]: Noticed a missing space.

File: docs/technical/code-snippets/boilerplate.resultprinter.example.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
+
```
Comment (Nikerabbit) [inline docs/technical/code-snippets/boilerplate.resultprinter.example.md]: Extra empty line?

File: docs/technical/code-snippets/boilerplate.resultprinter.example.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+This document contains an example for a `ResultPrinter` both for the PHP and JavaScript part and before diving into detailt make sure you have read ["Writing a result printer"](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/docs/hacking/writing.resultprinter.md).
```
Comment (Nikerabbit) [inline docs/technical/code-snippets/boilerplate.resultprinter.example.md]: detailt -> details?

File: docs/architecture/coding.conventions.md (hunk 1 above)
Comment (mwjames) [reply docs/architecture/coding.conventions.md]: Thanks. Done with #3768

File: docs/technical/code-snippets/boilerplate.resultprinter.example.md (hunk 2 above)
Comment (mwjames) [reply docs/technical/code-snippets/boilerplate.resultprinter.example.md]: Thanks. Done with #3768

File: docs/technical/code-snippets/boilerplate.resultprinter.example.md (hunk 3 above)
Comment (mwjames) [reply docs/technical/code-snippets/boilerplate.resultprinter.example.md]: Thanks. Done with #3768

**Pull request merged**